src/test/java/StringManipulationTest.java

```diff
@@ -19,130 +21,179 @@ public void tearDown() {
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(4, 3),
+                "start word allowed to be after end word");
     }
-    @Test
-    public void testGeSubStrings3() {
-        fail("Not yet implemented");
+    @Test // Tests index out of bounds exception
+    public void testGetSubStrings3() {
+        manipulatedstring.setString("This is my string");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(3, 5),
+                "end word exceeds number of words");
+    }
+    @Test // Tests illegal argument exception with word index too low
+    public void testGetSubStrings4() {
+        manipulatedstring.setString("This is my string");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.getSubStrings(0, 3),
+                "word index allowed to be negative or 0");
     }
-    @Test
-    public void testGeSubStrings4() {
-        fail("Not yet implemented");
+    @Test // Tests getting only final substring
+    public void testGetSubStrings5() {
+        manipulatedstring.setString("This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(4, 4);
+
+        assertEquals("string", sStings[0]);
     }
-    @Test
-    public void testGeSubStrings5() {
-        fail("Not yet implemented");
+    @Test // Tests getting only first substring
+    public void testGetSubStrings6() {
+        manipulatedstring.setString("This");
+        String [] sStings = manipulatedstring.getSubStrings(1, 1);
+
+        assertEquals("This", sStings[0]);
     }
-    @Test
-    public void testGeSubStrings6() {
-        fail("Not yet implemented");
+    @Test // Tests getting all substrings
+    public void testGetSubStrings7() {
+        manipulatedstring.setString("This is my string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 4);
+
+        assertEquals("This", sStings[0]);
+        assertEquals("string", sStings[3]);
     }
 
-    @Test
+    @Test // Tests standard usage of restore string
     public void testRestoreString1()
     {
         manipulatedstring.setString("art");
-        int [] array;
-        array=new int[]{1,0,2};
-        String restoreString = manipulatedstring.restoreString(array);
-        assertEquals(restoreString, "rat");
+        String restoreString = manipulatedstring.restoreString(new int[] {1, 0, 2});
+        assertEquals("rat", restoreString);
     }
 
-    @Test
+    @Test // Tests index out of bounds for invalid indices passed
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("art");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.restoreString(new int[] {4, 5, 6}));
 
     }
 
-    @Test
+    @Test // Tests illegal argument exception for too many indices passed
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
```

Comment:
This test will only run the fail part.

Reply:
The - before the line indicates that it was removed from the last commit